linkerd/stack/tracing/src/lib.rs

```diff
@@ -175,10 +138,11 @@ where
     }
 
     fn call(&mut self, request: Req) -> Self::Future {
-        let _enter = self.span.enter();
+        let span = self.get_span();
+        let _enter = span.enter();
 
         trace!(?request, "service");
-        self.inner.call(request).instrument(self.span.clone())
+        self.inner.call(request).instrument(span.clone())
```

Comment:
similarly, if we update our `tracing` dep to v0.1.25, we could write
```rust
let span = self.get_span().entered();

trace!(?request, "service");
self.inner.call(request).instrument(span.exit())
```
avoiding the span ref bump/dec from cloning and dropping the span, and saving one line :)